src/lib/containers/DirectDownload.tsx

```diff
@@ -52,27 +52,43 @@ const DirectDownload: React.FunctionComponent<DirectFileDownloadProps> = props =
   }, [accessToken, inView])
 
   const getDownloadLink = async () => {
-    let preSignedURL
-    try {
-      if (associatedObjectType === FileHandleAssociateType.TableEntity) {
-        const files = await getTableEntityFileHandle(true)
-        preSignedURL = files.requestedFiles[0].preSignedURL!
-      } else {
-        const file = await getFileResult(fileEntity!, accessToken, false, true)
-        preSignedURL = file.preSignedURL
+      let preSignedURL
+      // SWC-5907: opening in the file must be strictly done in the same click event process (Safari only).
+      // https://stackoverflow.com/questions/6628949/window-open-popup-getting-blocked-during-click-event
+      const isSafari = navigator.userAgent.indexOf("Safari") != -1 && navigator.userAgent.indexOf("Chrome") == -1
+      let safariDownloadWindow:Window | null = null
+      if (isSafari) {
+        safariDownloadWindow = window.open("", "Safari Download", "toolbar=no,location=no,directories=no,status=no,menubar=no,scrollbars=yes,resizable=yes,popup,width=600,height=200")
+        safariDownloadWindow!.document.body.innerHTML = "Downloading file on Safari..."
+      }
+      try {
+        if (associatedObjectType === FileHandleAssociateType.TableEntity) {
+          const files = await getTableEntityFileHandle(true)
+          preSignedURL = files.requestedFiles[0].preSignedURL!
+        } else {
+          const file = await getFileResult(fileEntity!, accessToken, false, true)
+          preSignedURL = file.preSignedURL
+        }
+      } catch (e) {
+        console.log('Fail to get file download link: ', e)
       }
-    } catch (e) {
-      console.log('Fail to get file download link: ', e)
-    }
 
-    if (!preSignedURL) {
-      console.log('Fail to get file download link')
-    } else {
-      window.open(preSignedURL)
-      if (onClickCallback) {
-        onClickCallback()
+      if (!preSignedURL) {
+        safariDownloadWindow?.close()
+        console.log('Fail to get file download link')
+      } else {
+        if (isSafari && safariDownloadWindow) {
```

Comment:
I'm wondering if there could possibly be a race condition where we get here and the window hasn't opened yet, but probably not possible since JS is single-threaded.

Reply:
You're right in that the creation (and URL load) of the Window are asynchronous.  But window.open() actually returns a WindowProxy that has all features immediately available.
https://developer.mozilla.org/en-US/docs/Web/API/Window/open